network/server.cpp

```diff
@@ -1,62 +1,77 @@
 #include "../include/network/server.hpp"
+#include "../include/network/DelayEstimator.hpp"
 
 /* PRIVATE */
 void NetworkServer::newConnection() {
   QTcpSocket *clientConnection = server->nextPendingConnection();
-  connect(clientConnection, &QAbstractSocket::disconnected, clientConnection, &QObject::deleteLater);
+  // The delete later is a problem for classes having a pointer to the client Socket (DelayEstimator)
+  // connect(clientConnection, &QAbstractSocket::disconnected, clientConnection, &QObject::deleteLater);
```

Comment:
You can connect `disconnected` to `DelayConnection`'s destructor, and remove the `DelayConnection` from the `DelayEstimator` from there.

We don't care a lot about memory leaks, so maybe just leave that as is.